lib/deadpool-cyclone/src/instance/cyclone/local_uds.rs

```diff
@@ -90,12 +103,17 @@ impl Instance for LocalUdsInstance {
     type Error = LocalUdsInstanceError;
 
     async fn terminate(mut self) -> result::Result<(), Self::Error> {
-        if !self.watch_shutdown_tx.is_closed() && self.watch_shutdown_tx.send(()).is_err() {
-            debug!("sent watch shutdown but receiver was already closed");
+        match self.runtime.runtime_strategy {
+            LocalUdsRuntimeStrategy::LocalProcess => {
+                if !self.watch_shutdown_tx.is_closed() && self.watch_shutdown_tx.send(()).is_err() {
+                    debug!("sent watch shutdown but receiver was already closed");
+                }
+                return self.runtime.terminate().await;
+            }
+            LocalUdsRuntimeStrategy::LocalDocker => {
+                return self.runtime.terminate().await;
+            }
```

Comment:
I'd have to check what `terminate()` does, but you should not have to return them since the function returns `result::Result<(), Self::Error>`. You can likely use the "`try_intro`" operator ("`?`") instead.